packages/ui/src/components/TokenMedia/index.tsx

```diff
@@ -0,0 +1,296 @@
+import React, {
+  CSSProperties,
+  FC,
+  HTMLAttributes,
+  ReactElement,
+  SyntheticEvent,
+  useContext,
+  useState,
+  useRef,
+  LegacyRef,
+} from 'react'
+import { useTokens } from '../../hooks'
+import { useModelViewer } from '../../hooks'
+import { ThemeContext } from '../../ReservoirKitProvider'
+import { Box } from '../../primitives'
+import MediaPlayButton from './MediaPlayButton'
+import { useMeasure } from '@react-hookz/web'
+import TokenFallback from './TokenFallback'
+
+type MediaType =
+  | 'mp4'
+  | 'mp3'
+  | 'wav'
+  | 'gltf'
+  | 'glb'
+  | 'png'
+  | 'jpeg'
+  | 'jpg'
+  | 'svg'
+  | 'gif'
+  | 'html'
+  | 'other'
+  | undefined
+
+export const extractMediaType = (
+  token?: RequiredTokenProps
+): MediaType | null => {
+  let extension: string | null = null
+  if (token?.media) {
+    const pieces = token.media.split('/')
+    const file =
+      pieces && pieces[pieces.length - 1] ? pieces[pieces.length - 1] : null
+    const matches = file ? file.match('(\\.[^.]+)$') : null
+    extension = matches && matches[0] ? matches[0].replace('.', '') : null
+  }
+  return (extension as MediaType) ? (extension as MediaType) : null
+}
+
+type Token = NonNullable<
+  NonNullable<ReturnType<typeof useTokens>['data']>['0']
+>['token']
+
+type RequiredTokenProps = Pick<
+  NonNullable<Token>,
+  'image' | 'media' | 'collection' | 'tokenId'
+>
+
+type Props = {
+  token?: RequiredTokenProps
+  preview?: boolean
+  style?: CSSProperties
+  className?: string
+  modelViewerOptions?: any
+  videoOptions?: HTMLAttributes<HTMLVideoElement>
+  audioOptions?: HTMLAttributes<HTMLAudioElement>
+  iframeOptions?: HTMLAttributes<HTMLIFrameElement>
+  fallback?: (mediaType: MediaType | null) => ReactElement | null
+  onError?: (e: Event) => void
+  onRefreshToken?: () => void
+}
+
+const TokenMedia: FC<Props> = ({
+  preview,
+  token,
+  style,
+  className,
+  modelViewerOptions = {},
+  videoOptions = {},
+  audioOptions = {},
+  iframeOptions = {},
+  fallback,
+  onError = () => {},
+  onRefreshToken = () => {},
+}) => {
+  const mediaRef = useRef<HTMLAudioElement | HTMLVideoElement>(null)
+  const themeContext = useContext(ThemeContext)
+  const borderRadius: string =
```

Comment:
can this be

`let borderRadius: string = themContext?.radii?.borderRadius?.value || '0'`